File: frontend/lib/models/object_status.dart
enum ObjectStatus {
  active('Aktif', 'ACTIVE'),
  needsMaintenance('Bakım Gerekli', 'NEEDS_MAINTENANCE'),
  broken('Kırık', 'BROKEN'),
  removed('Kaldırıldı', 'REMOVED');

  final String displayName;
  final String apiValue;

  const ObjectStatus(this.displayName, this.apiValue);

  static ObjectStatus fromApiValue(String value) {
    return ObjectStatus.values.firstWhere(
      (e) => e.apiValue == value,
      orElse: () => throw ArgumentError('Bilinmeyen Cisim Türü: $value'),
    );
  }
}

File: backend/cografi-nesne-takip/src/main/java/com/kutalmis/cografi_nesne_takip/entity/ObjectStatus.java
package com.kutalmis.cografi_nesne_takip.entity;

public enum ObjectStatus {
    ACTIVE("Aktif"),
    NEEDS_MAINTENANCE("Bakım Gerekli"),
    BROKEN("Kırık"),
    REMOVED("Kaldırıldı");

    private final String displayName;

    ObjectStatus(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
}